### PR TITLE
Add libdeflate to the wheel build.

### LINKED ISSUE #224: libdeflate
cyvcf2 is substantially faster on BCF when libdeflate is used. 
see: https://github.com/brentp/vcf-bench

this takes the time from ~8 seconds to ~4 seconds.

would be nice to get this into the default build

### PR DESCRIPTION
**Pull request by grahamgower**
Closes #224.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -54,13 +54,19 @@ def no_cythonize(extensions, **_ignore):
...
 sources.append('cyvcf2/helpers.c')
 
+extra_objects = []
+LIBDEFLATE = os.getenv("LIBDEFLATE")
+if LIBDEFLATE is not None:
+    extra_objects.append(LIBDEFLATE)
+
 extensions = [Extension("cyvcf2.cyvcf2",
                         ["cyvcf2/cyvcf2.pyx"] + sources,
                         libraries=['z', 'bz2', 'lzma', 'curl', 'ssl'] + (['crypt'] if platform.system() != 'Darwin' else []),
```
Comment (brentp) [inline setup.py:64]: why not just add `deflate` to libraries if LIBDEFLATE is not None?

Comment (grahamgower) [reply setup.py:64]: That does dynamic linking on my system. Does the ciwheelbuild pull that in dynamically or statically? Maybe it just works there?

**Comment by grahamgower** [comment]
It works! If you're happy with this @brentp, I'll back out the `on: pull_request:` and squash the commits.

**Comment by brentp** [comment]
LGTM. let me know when you're ready to merge.

**Comment by grahamgower** [comment]
Ok, this is good to go @brentp.

**Comment by brentp** [comment]
excellent!!! thank you.

**Pull request merged**